src/sc_client/constants/sc_type/sc_type.py

```diff
@@ -0,0 +1,198 @@
+
+    def is_membership_arc(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_MEMBERSHIP_ARC) != 0
+
+    def is_link(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_LINK) == bitmasks.SC_TYPE_NODE_LINK
+
+    def is_const(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_CONST) != 0
+
+    def is_var(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_VAR) != 0
+
+    def is_pos(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_POS_ARC) == bitmasks.SC_TYPE_POS_ARC
+
+    def is_neg(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NEG_ARC) == bitmasks.SC_TYPE_NEG_ARC
+
+    def is_fuz(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_FUZ_ARC) == bitmasks.SC_TYPE_FUZ_ARC
+
+    def is_perm(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_PERM_ARC) == bitmasks.SC_TYPE_PERM_ARC
+
+    def is_temp(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_TEMP_ARC) == bitmasks.SC_TYPE_TEMP_ARC
+
+    def is_actual(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_ACTUAL_ARC) == bitmasks.SC_TYPE_ACTUAL_ARC
+
+    def is_inactual(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_INACTUAL_ARC) == bitmasks.SC_TYPE_INACTUAL_ARC
+
+    def is_tuple(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_TUPLE) == bitmasks.SC_TYPE_NODE_TUPLE
+
+    def is_structure(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_STRUCTURE) == bitmasks.SC_TYPE_NODE_STRUCTURE
+
+    def is_struct(self) -> bool:
+        warnings.warn("ScType `is_struct` method is deprecated. Use `is_structure` method instead.", DeprecationWarning)
+        return self.is_structure()
+
+    def is_role(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_ROLE) == bitmasks.SC_TYPE_NODE_ROLE
+
+    def is_no_role(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_NO_ROLE) == bitmasks.SC_TYPE_NODE_NO_ROLE
+
+    def is_norole(self) -> bool:
+        warnings.warn("ScType `is_norole` method is deprecated. Use `is_no_role` method instead.", DeprecationWarning)
+        return self.is_no_role()
+
+    def is_class(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_CLASS) == bitmasks.SC_TYPE_NODE_CLASS
+
+    def is_superclass(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_SUPERCLASS) == bitmasks.SC_TYPE_NODE_SUPERCLASS
+
+    def is_material(self) -> bool:
+        return (self.value & bitmasks.SC_TYPE_NODE_MATERIAL) == bitmasks.SC_TYPE_NODE_MATERIAL
+
+    def is_valid(self) -> bool:
+        return self.__bool__()
+
+    def is_equal(self, other: ScType) -> bool:
+        return self.__eq__(other)
+    
+    def _is_not_compatible_by_mask(self, new_type: ScType, mask):
+        subtype = self.value & mask
+        new_subtype = new_type.value & mask
+        return subtype != bitmasks.SC_TYPE_UNKNOWN and subtype != new_subtype
```

Comment:
this method can be static, it is mostly used right after constructing two new ScType variables but it is not necessary, ints can be passed here

Reply:
What are the advantages of the static method over the current way? It doesn't provide anything here

Reply:
it is always called right after constructing of new objects, so it'll use less statements to do same task
![image](https://github.com/user-attachments/assets/e00b6e64-6e59-4303-9816-a1b674ee69b9)

Reply:
`it'll use less statements to do same task` What do you mean?

Reply:
Constructors and destructors won't be called so at least 4 statements per check will not be executed
Ints can be compared in static method without the creation of ScTypes

Reply:
What does that do?

Reply:
Fixed, ok